agent/algorithm/docker/docker.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Ultraviolet
+// SPDX-License-Identifier: Apache-2.0
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/ultravioletrs/cocos/agent/algorithm"
+	"github.com/ultravioletrs/cocos/agent/events"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	containerName       = "agent_container"
+	DockerRunCommand    = "python3 /cocos/algorithm.py"
```

Comment:
This is not clear to me, what is this const? Does that mean that we can run only Python algorithms? Should we just run a docker, whatever the binary is inside, and it is a docker creator's obligation to ensure that the container will boot properly, that the binary (or script, or whatever) will run inside and that it will find datasets mapped on the volume

Reply:
it is just the default, but the user can supply their own command to run in docker

Reply:
I have changed the code so that the docker start command must be a part of the docker image. The `datasets` and `results` are left as options in the CLI. They are there because the `python`, `wasm` and `bin` will not work if the datasets and results dirs are `/datasets` and `/results`. They need to be relative for those algorithms.